### PR TITLE
Switch Linux HDID code to Qt shenanigans

### LINKED ISSUE #856: Improve HDID calculation on Linux
Currently, the first uncommented line from /etc/fstab is read and searched for a "UUID=" portion. Not only is this very brittle, as it does not work properly for systems which use block device filenames rather than UUIDs, it is also very prone to manipulation; changing the ID on Linux is as simple as adding a bogus entry to the fstab.

There are myriad better ways to improve this system. The biggest challenge is arguably determining which filesystem is fixed and can be used to uniquely identify the solution; the trivial, and likely best solution is to choose the root partition. However, it is worth noting, that the root partition does not necessarily have to be backed by a physical block device, and may not be the best way to identify a computer uniquely.

That said, determining the UUID of the root partition is likely the most effective parallel to the Windows method. This is liable to change upon re-installing the OS, but it is less brittle than the current solution. A `system()` call to the `mount` utility, or parsing the fstab, can both determine the block device (and trivially the UUID) of the root partition.

Another approach would be to use another source of information to determine a unique fingerprint. A few examples may be:

### /etc/machine-id
This file is created by all distros that I am aware of at install time. It is very simple to read and does not change. However, a savvy user can easily edit this file, and it is not persistent between installations.

### Network adapter HWAddrs
These are tied to hardware, but once again, a savvy user can change this, and may not be as unique as is needed for these purposes

### Hash of enumerated hardware data (e.g. lspci)
This data is largely static and difficult to change without modifying the configuration of the hardware, however, any given setup has a distinct change of lacking uniqueness. Furthermore, simple hardware changes may invalidate the fingerprint.

These are just a few ideas off of the top of my head, and I hope that more can be discussed in this issue. Furthermore, I think all of these proposals are flawed in one way or another, but make no mistake: the current system is too error-prone to remain in use, and I think any of these would be an improvement. My personal choice is /etc/machine-id, but I'd love to see some discussion on this topic.

### PR DESCRIPTION
**Pull request by Salanto**
Applies longbytes suggestion from #856 

closes #856

### CODE REVIEW

**Comment by stonedDiscord** [comment]
does it actually give you a valid hdid on linux

**Comment by Salanto** [comment]
It joins servers perfectly fine on Linux, including Vanilla. Akashi receives a valid HDID too.
One would need to test it on Arch perhaps as those had issues for sure.

**Comment by stonedDiscord** [comment]
valid or the default one?

**Comment by Salanto** [comment]
Valid one, obviously. It wouldn't work if I got the default one, would it?

**Comment by stonedDiscord** [comment]
it would work with the default one, until someone on linux gets banned

**Comment by Salanto** [comment]
Well, that is the curse of default. Not exactly able to prevent that. It would at least clue us in when their HDID is not able to be retrieved.

File: src/hardware_functions.cpp [hunk 1]
```diff
@@ -50,30 +50,18 @@ QString get_hdid()
...
-
-    if (line.startsWith("UUID")) {
-      QStringList line_elements = line.split("=");
+  machineId = QSysInfo::machineUniqueId();
 
-      if (line_elements.size() > 1)
-        return line_elements.at(1).left(23).trimmed();
-    }
+  if (machineId.isEmpty()) {
+    return "gxsps32sa9fnwic92mfbs2";
```
Comment (oldmud0) [inline src/hardware_functions.cpp:62]: i like how only a few letters and digits were changed. totally random

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -130,18 +130,18 @@ jobs:
...
-          cp ./bass/x64/libbassmidi.so ./bin/
+          cp ./bass/libs/x86_64/libbassmidi.so ./lib/
+          cp ./bass/libs/x86_64/libbassmidi.so ./bin/
 
           curl http://www.un4seen.com/files/bassopus24-linux.zip -o bassopus.zip
           unzip -d bass -o bassopus.zip
-          cp ./bass/x64/libbassopus.so ./lib/
-          cp ./bass/x64/libbassopus.so ./bin/
+          cp ./bass/libs/x86_64/libbassopus.so ./lib/
+          cp ./bass/libs/x86_64/libbassopus.so ./bin/
```
Comment (oldmud0) [inline .github/workflows/build.yml:144]: This version bump can ride this PR, I guess

**Review by oldmud0 — APPROVED** [review]

**Pull request merged**